tools/sof-dump-status.py

```diff
@@ -294,6 +346,11 @@ def loadDAPM(self, filter = "all"):
                     self.dapm['name_lst'].append("'%s/%s'" %(path_name ,fname.name))
 
 if __name__ == "__main__":
+    def read_file(filepath):
```

Comment:
Why not at the top-level?

Reply:
Because I believed it would be a good idea to have all the functions grouped together but if that's not needed then I see no reason why it shouldn't be placed at the top of the module.

Reply:
I think hiding functions away from the top-level is a bit unusual in Python (except for closures of course). For instance there is no `private` keyword https://docs.python-guide.org/writing/style/#we-are-all-responsible-users

This function is really super-generic, could be used anywhere.

Again I appreciate the consistency effort (I really do!) but having a giant `main()` function with all other functions inside seems like it was a pure mistake and waste of whitespace / indentation with no apparent benefit.